apps/stream-consumer/src/email/email.service.ts

```diff
@@ -29,4 +35,59 @@ export class EmailService {
       `,
     });
   }
+
+  async sendOrderApprovalNeeded(event: PurchaseOrderCreatedEvent) {
+    if (!event.data.approvers?.length) {
+      this.logger.warn(
+        `received approval event with no approvers. purchase order # ${event.data.purchaseOrderId}`,
+      );
+      return;
+    }
+
+    await this.transporter.sendMail({
+      from: process.env.FROM_EMAIL,
+      to: event.data.approvers,
+      subject: `Purchase Order # ${event.data.purchaseOrderId} Needs Approval`,
+      html: `
+        <h2>Purchase Order Needs Approval</h2>
+        <p>The details are below:</p>
+        <div>Vendor name: ${event.data.vendorName}</div>
+        <div>Total Quantity: ${event.data.totalQuantity}</div>
+        <div>Total Cost: $${event.data.totalCost}</div>
+        <div>Expected Delivery Date: ${event.data.expectedDeliveryDate}</div>
+        <div>Created by: ${event.data.createdBy}</div>
+        <div>Line Item Count: ${event.data.items.length}</div>
+        <br><br>
+        <p><a href="http://localhost:4200/purchase-orders/${event.data.purchaseOrderId}">Click here to review</a></p>
+      `,
```

Comment:
Sends details about the purchase order in the email when requesting approval. I decided against added "magic links" to let an admin approve/deny straight from the email, because I felt there was too much info that was needed to make that decision and an email wasn't the right medium for it.